src/features/project/Project.service.ts

```diff
@@ -14,9 +15,33 @@ export class ProjectService {
     const saveProject = await this.projectDao.save(project);
 
     const users = await this.memberDao.find();
+
     for (const user of users) {
       this.projectUtils.sendMailToUsers(user);
     }
+
     return { id: saveProject.id };
   }
+
+  async getProjects(category: string) {
+    const projects = await this.projectDao.find();
+    const result = [];
+
+    for (let i = 0; i < projects.length; i++) {
+      const project = projects[i] as GetProjectDetails;
+      const today = DateTime.now().toFormat('yyyy-MM-dd');
+      if (category == 'previous' && today > project.endDate) {
+        project.status = 'COMPLETED';
+        result.push(project);
+      } else if (category == 'upcoming' && today < project.startDate) {
+        project.status = 'UPCOMING';
+        result.push(project);
+      } else if (category == 'current' && today <= project.endDate && today >= project.startDate) {
+        project.status = 'ONGOING';
+        result.push(project);
```

Comment:
Everything is working perfectly. Here `result.push(project)` is repeating 3 times. You can extract that piece of if blocks as below.

```
for (let i = 0; i < projects.length; i++) {
      const project = projects[i] as GetProjectDetails;
      const today = DateTime.now().toFormat('yyyy-MM-dd');

      if (category == 'previous' && today > project.endDate) {
        project.status = 'COMPLETED';
      } else if (category == 'upcoming' && today < project.startDate) {
        project.status = 'UPCOMING';
      } else if (category == 'current' && today <= project.endDate && today >= project.startDate) {
        project.status = 'ONGOING';
      }
      
      result.push(project);
    }
```

Reply:
Talked with ani.